### PR TITLE
Allow specifying attributes for `RequestMatcher`

### LINKED ISSUE #17377: [SecurityBundle] Allow specifying attributes for `RequestMatcher`
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#45907
| PR author(s) | @freiondrej-lmc
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by mdoutreluingne**
fixes #17377 

Could you help me with the XML and PHP formats.

- [x] Add YAML example format
- [x] Add XML example format
- [x] Add PHP example format

### CODE REVIEW

File: security/access_control.rst [hunk 1]
```diff
@@ -30,11 +30,17 @@ options are used for matching:
 * ``host``: a regular expression
 * ``methods``: one or many HTTP methods
 * ``request_matcher``: a service implementing ``RequestMatcherInterface``
+* ``attributes``: is an array, which can be used to specify attributes of the query
```
Comment (wouterj) [inline security/access_control.rst]: In the other items, we document what type of match is expected. Are attributes matched using a regular expression, or is it always an exact match?

I think we should also link "request attributes" to e.g. https://symfony.com/doc/current/components/http_foundation.html#accessing-request-data if people don't know what request attributes are. You can do so using the ``:ref:`request attributes <accessing-request-data>` `` syntax.

E.g. something along the lines of: ``one or more :ref:`request attributes <accessing-request-data>` that must match exactly `` (not sure if it's exact match)

**Review by wouterj — COMMENTED** [review]
Thank you!

The other formats would look like this:
```xml
<rule role="ROLE_ADMIN">
    <attribute key="_route">admin</attribute>
</rule>

<rule route="admin" role="ROLE_ADMIN"/>
```

(btw, you were actually right - XML was not yet supported for this setting, I've contributed a fix: https://github.com/symfony/symfony/pull/47959 )

```php
$security->accessControl()
    ->roles(['ROLE_ADMIN'])
    ->attributes(['_route' => 'admin'])
;

$security->accessControl()
    ->roles(['ROLE_ADMIN'])
    ->route('admin')
;
```

File: security/access_control.rst (hunk 1 above)
Comment (mdoutreluingne) [reply security/access_control.rst]: Thanks for the help Wouter.

I may have misunderstood but for the PHP examples, I don't see the methods `route()` and `attributes()` in the `AccessControlConfig` class.

Yes, very good idea for the link "request attributes", it is added. I don't see a match using a regular expression in the [PR](https://github.com/symfony/symfony/pull/45907/files), so I think it's an exact match

Comment (wouterj) [reply security/access_control.rst]: Thanks for the quick reaction :)

I just checked locally, and I can see the 2 methods in the `AccessControlConfig` class. Maybe you need to update `symfony/security-bundle` to the latest `6.2.*@dev` version or regenerate the Symfony cache?

Comment (mdoutreluingne) [reply security/access_control.rst]: It was the cache... Thanks! :)

**Review by wouterj — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
What does WCM in the title means?

**Comment by mdoutreluingne** [comment]
[WCM](https://symfony.com/doc/current/contributing/documentation/overview.html#what-if-i-want-to-submit-my-work-without-fully-finishing-it) means 'Waiting Code Merge', Wouter has created a [PR ](https://github.com/symfony/symfony/pull/47959)to add the XML format

**Comment by wouterj** [comment]
Ready to go 🚀

**Comment by OskarStark** [comment]
Thank you Maxime.

**Pull request merged**